src/ipc/ipc4/handler.c

```diff
@@ -965,6 +965,37 @@ void ipc_msg_reply(struct sof_ipc_reply *reply)
 	ipc_compound_msg_done(in.primary.r.type, reply->error);
 }
 
+void ipc_msg_send_status(void)
+{
+	char *data = ipc_get()->comp_data;
+	int count = 10;
+	int ret;
+
+	/* copy mailbox data to message */
```

Comment:
Why you need to re-implement the `ipc_send_msg()`?
`ipc_msg_send(&msg_reply, data, true);` would do the same, no?

Reply:
No, two issues. First it will require ipc_lock which will result to dead lock since the framework gets the lock
Second it may save the reply msg to msg queue not send it immediately,  needs anther interrupt to trigger queue msg send. But at this time host is waiting for reply msg and there is no irq, so reply msg will be never sent

Reply:
sure.